core/src/test/java/io/confluent/rest/ErrorMessageIntegrationTest.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2014 - 2023 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParseException;
+import io.confluent.rest.entities.ErrorMessage;
+import io.confluent.rest.exceptions.JsonParseExceptionMapper;
+import java.util.Properties;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+@Tag("IntegrationTest")
+public class ErrorMessageIntegrationTest {
+
+  private Server server;
+  private Client client;
+
+  @BeforeEach
+  public void setUp(TestInfo testInfo) throws Exception {
+    Properties props = new Properties();
+    TestRestConfig config = new TestRestConfig(props);
+    TestApplication app = new TestApplication(config,
+        testInfo.getDisplayName().contains("WithJsonMediaType"));
+    app.createServer();
+    server = app.createServer();
+    server.start();
+    client = ClientBuilder.newClient(app.resourceConfig.getConfiguration());
+  }
+
+  @AfterEach
+  public void tearDown() throws Exception {
+    server.stop();
+    server.join();
+    client.close();
+  }
+
+  @Test
+  @DisplayName("testOPTIONS_WithoutMediaType_return500")
```

Comment:
Why test this path?  If we've got the fix right then all endpoints return properly (ie not 500) because we've set mediaType everywhere it's needed.

Reply:
This test is just demonstration that without MediaType, we would get 500, I put a comment on this